vms/proposervm/block.go

```diff
@@ -328,7 +361,102 @@ func (p *postForkCommonComponents) verifyBlockDelay(
 	return delay, nil
 }
 
-func (p *postForkCommonComponents) shouldBuildUnsignedBlock(
+func (p *postForkCommonComponents) verifyPostDurangoBlockDelay(
+	ctx context.Context,
+	parentTimestamp time.Time,
+	parentPChainHeight uint64,
+	blk *postForkBlock,
+) error {
+	var (
+		blkTimestamp = blk.Timestamp()
+		blkHeight    = blk.Height()
+		proposerID   = blk.Proposer()
+	)
+
+	expectedProposerID, err := p.vm.Windower.ExpectedProposer(
+		ctx,
+		blkHeight,
+		parentPChainHeight,
+		proposer.TimeToSlot(parentTimestamp, blkTimestamp),
+	)
+	if err != nil {
+		p.vm.ctx.Log.Error("unexpected block verification failure",
+			zap.String("reason", "failed to calculate expected proposer"),
+			zap.Stringer("blkID", blk.ID()),
+			zap.Error(err),
+		)
+		return err
+	}
+	if expectedProposerID != proposerID {
+		return errUnexpectedProposer
+	}
+
+	return nil
+}
+
+func (p *postForkCommonComponents) shouldBuildBlockPostDurango(
```

Comment:
nit: I think `scheduleBuildBlockPostDurango` is a better name for what this actually is

`shouldBuild*` at least implies a `bool` is returned?

Reply:
hmmm I don't know what to call this function... I don't like `scheduleBuildBlockPostDurango` - because it only schedules the build block if it returns an error...

Reply:
I'm going to add a TODO to restructure this function after durango activates... I think it'll be easier to clean up the code during that than now